src/fetch/period_slippage.py

```diff
@@ -28,22 +31,43 @@ def prepend_to_sub_query(query: str, table_to_add: str) -> str:
 
 def add_token_list_table_to_query(original_sub_query: str) -> str:
     """Inserts the token_list table right after the WITH statement into the sql query"""
-    token_list = get_trusted_tokens_from_url(ALLOWED_TOKEN_LIST_URL)
-    sql_query_for_allowed_token_list = allowed_token_list_query(token_list)
-    return prepend_to_sub_query(original_sub_query, sql_query_for_allowed_token_list)
+    token_list = fetch_trusted_tokens()
+    allowed_tokens_query = allowed_token_list_query(token_list)
+    return prepend_to_sub_query(original_sub_query, allowed_tokens_query)
 
 
-def slippage_query(dune: DuneAnalytics) -> str:
-    """Constructs our slippage query by joining sub-queries"""
-    path = "./queries/slippage"
-    slippage_sub_query = dune.open_query(
-        File("subquery_batchwise_internal_transfers.sql", path).filename()
-    )
-    select_slippage_query = dune.open_query(
-        File("select_slippage_results.sql", path).filename()
-    )
+class QueryType(Enum):
+    """
+    Determines type of slippage data to be fetched.
+    The slippage subquery allows us to select from either of the two result tables defined here.
+    """
+
+    PER_TX = "results_per_tx"
+    TOTAL = "results"
+
+    def __str__(self):
+        return self.value
+
+
+def slippage_query(query_type: QueryType = QueryType.TOTAL) -> str:
+    """
+    Constructs our slippage query by joining sub-queries
+    Default query type input it total, but we can request
+    per transaction results for testing
+    """
+
+    slippage_sub_query = DuneAnalytics.open_query("./queries/period_slippage.sql")
+    select_statement = f"""
+    select *, 
+        usd_value / (select price from eth_price) * 10 ^ 18 as eth_slippage_wei 
+    from {query_type}
+    """.strip()
```

Comment:
This removes the need to pass the {{ResultTable}} as a query parameter (since we are dynamically using the string right here)